libs/auth-api-lib/src/lib/resources/delegation-resources.service.ts

```diff
@@ -94,4 +169,101 @@ export class DelegationResourcesService {
         children: groupChildren.get(node.name),
       }))
   }
+
+  private wrapApiScopeQuery<T extends ApiScopeQueryOptions>(
+    user: User,
+    prefix: string,
+    queryOptions: T,
+  ): T {
+    const skipScopes = this.delegationConfig.customScopeRules
+      .filter((scopeRule) => !this.scopeRuleMatchesUser(user, scopeRule))
+      .map((scopeRule) => scopeRule.scopeName)
+
+    let where = queryOptions.where ?? { [Op.and]: [] }
+    if (!Array.isArray(where[Op.and as never])) {
+      where = { [Op.and]: [where] }
+    }
+    const whereAnd = where[Op.and as never] as Array<WhereOptions>
+    whereAnd.push({
+      allowExplicitDelegationGrant: true,
+      enabled: true,
+      ...(skipScopes.length > 0 ? { name: { [Op.notIn]: skipScopes } } : {}),
+      [Op.or]: [
+        { isAccessControlled: { [Op.ne]: true } },
+        { [`$${prefix}apiScopeUserAccesses.scope$`]: { [Op.ne]: null } },
+      ],
+    })
+
+    let include = queryOptions.include ?? []
+    if (!Array.isArray(include)) {
+      include = [include]
+    }
+    include.push({
+      attributes: [],
+      model: ApiScopeUserAccess,
+      where: {
+        nationalId: user.nationalId,
+      },
+      required: false,
+    })
+
+    // Currently only supporting access control for company delegations.
+    if (user.delegationType && user.actor) {
+      const delegationOr: Array<WhereOptions<ApiScope>> = []
+      whereAnd.push({ [Op.or]: delegationOr })
+      if (user.delegationType.includes('ProcurationHolder')) {
+        delegationOr.push({ grantToProcuringHolders: true })
+      }
+      if (user.delegationType.includes('LegalGuardian')) {
+        delegationOr.push(literal('false'))
+      }
+      if (user.delegationType.includes('PersonalRepresentative')) {
+        delegationOr.push(literal('false'))
+      }
```

Comment:
Is this just to be explicit about that `LegalGuardian` and `PersonalRepresentative` will not result in any scopes?

Reply:
Yeah, this was tricky. There are currently no plans to allow legal guardians and personal representatives to create delegations. They don't have scope-level access to this API so this should never trigger. But I was unsure how to leave the logic and tests around this. Also, correctly figuring out the correct scopes for personal representatives require extra joins that I didn't want to get into.

So I decided to leave it semi-explicit like this, though there is nothing there to stop individuals with custom delegations for other individuals to grant access to those scopes.

I'll add a comment explaining the thinking there.

Reply:
Actually, decided to remove this, only handle company delegations and throw an error for individual delegations.